### PR TITLE
fix gkehub test cases

### PR DESCRIPTION
**Pull request by ziyue-101**
### Change description

Fix issues
* service account name was too long
* feature resource should not be cleaned up when running tests concurrently
* features under spec.configmanagement should not be empty as it would rejected by the GCP API.

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by ziyue-101** [comment]
/hold

I would like to merge https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2230 first

File: pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/fullacmgkehubfeaturemembership/update.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
       sourceFormat: hierarchy
       git:
         gcpServiceAccountRef:
-          name: iamserviceaccount-${uniqueId} 
+          name: sa-${uniqueId}
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/fullacmgkehubfeaturemembership/update.yaml:33]: Is this change for the gsa length limitation?

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

One comment, otherwise looks good!

File: pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/fullacmgkehubfeaturemembership/update.yaml (hunk 1 above)
Comment (ziyue-101) [reply pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/fullacmgkehubfeaturemembership/update.yaml:33]: Yes

**Pull request merged**